store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/operation/handler/spark/utilities/WriteData.java

```diff
@@ -56,28 +59,47 @@ public WriteData(final Function<String, String> groupToDirectory, final Schema s
     public void call(final Iterator<Element> elements) throws Exception {
         final SchemaUtils schemaUtils = new SchemaUtils(Schema.fromJson(schemaAsJson));
         final int partitionId = TaskContext.getPartitionId();
+        final long taskAttemptId = TaskContext.get().taskAttemptId();
         final Map<String, ParquetWriter<Element>> groupToWriter = new HashMap<>();
+        final Map<String, Path> groupToWriterPath = new HashMap<>();
+
         for (final String group : schemaUtils.getGroups()) {
-            groupToWriter.put(group,
-                    buildWriter(group, groupToDirectory.get(group) + "/input-" + partitionId + ".parquet", schemaUtils));
+            groupToWriterPath.put(group, new Path(groupToDirectory.get(group) + "/input-" + partitionId + "-" + taskAttemptId + ".parquet"));
+            groupToWriter.put(group, buildWriter(group, groupToWriterPath.get(group), schemaUtils));
         }
         while (elements.hasNext()) {
             final Element element = elements.next();
             final String group = element.getGroup();
             groupToWriter.get(group).write(element);
         }
-        LOGGER.info("Finished writing elements for partition id {} to {}", partitionId, groupToDirectory);
+        LOGGER.info("Finished writing elements for partition id {} and task attempt id {} to {}",
+                partitionId, taskAttemptId, groupToDirectory);
         for (final ParquetWriter<Element> writer : groupToWriter.values()) {
             LOGGER.debug("Closing writer {}", writer);
             writer.close();
         }
+
+        LOGGER.info("Renaming output files from {} to {}",
```

Comment:
Not sure this is in the scope of the issue

Reply:
It's not, but in fixing this issue I found it useful to log that so left it in. I don't see any harm in adding it in this issue.

Reply:
Sorry - my comment wasn't very clear. I don't mind the logging but I was really on about renaming the files to remove the attempt Id from the file name. I was wondering whether it was actually nessecary.

Reply:
Yes, renaming them is critical to fixing the issue. A unique filename needs to initially be created for each call of the function to avoid the failed to create file error but then the result needs to be renamed to the correct file name. Whichever call of the function renames it first wins, other calls just don't do the rename.